AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -2315,31 +2335,53 @@ function New-ADAccountForStorageAccount {
             -Filter { ServicePrincipalNames -eq $spnValue } `
             -Server $Domain
 
-    if ($null -ne $computerSpnMatch -or $null -ne $userSpnMatch) {
-        Write-Error -Message "An AD object with a Service Principal Name of $spnValue already exists within AD. This might happen because you are rejoining a new storage account that shares names with an existing storage account, or if the domain join operation for a storage account failed in an incomplete state. Delete this AD object (or remove the SPN) to continue. See https://docs.microsoft.com/azure/storage/files/storage-troubleshoot-windows-file-connection-problems for more information." -ErrorAction Stop
+    if (($null -ne $computerSpnMatch) -and ($null -ne $userSpnMatch)) {
+        Write-Error -Message "There are already two AD objects with a Service Principal Name of $spnValue in domain $Domain." -ErrorAction Stop
+    } elseif (($null -ne $computerSpnMatch) -or ($null -ne $userSpnMatch)) {
+        if (-not $OverwriteExistingADObject) {
+            Write-Error -Message "An AD object with a Service Principal Name of $spnValue already exists within AD. This might happen because you are rejoining a new storage account that shares names with an existing storage account, or if the domain join operation for a storage account failed in an incomplete state. Delete this AD object (or remove the SPN) to continue or specify a switch -OverwriteExistingADObject when calling this cmdlet. See https://docs.microsoft.com/azure/storage/files/storage-troubleshoot-windows-file-connection-problems for more information." -ErrorAction Stop

```

Comment:
Nit: What if there's an existing user object but the customer wants to create a computer object? Should we give a different error here to the user?

Reply:
Is it supported to have a user object and a computer object for the same SPN?

Reply:
No it's not supported - I imagine the AD would have trouble deciding what AD object to issue the service ticket for.  We should block creation of a duplicate (but different type) of object and give a customer an error message about this.